code/modules/power/cable.dm

```diff
@@ -415,3 +412,208 @@
 		logTheThing(LOG_STATION, user, "lays a cable[powered == 1 ? " (powered when connected)" : ""] at [log_loc(src)].")
 
 	return
+
+/// a cable spawner which can spawn multiple cables to connect to other cables around it.
+/obj/cablespawner
+	name = "power cable spawner"
+	icon = 'icons/obj/power_cond.dmi'
+	icon_state = "superstate"
+	layer = CABLE_LAYER
+	plane = PLANE_NOSHADOW_BELOW
+	color = "#DD0000"
+	// this would make it connect to the centre, for like terminals and whatnot
+	// has to be var edited because lazy
+	var/override_centre_connection = FALSE
+	var/cable_type = /obj/cable
+	// some bit flags for 8 bit directions
+	var/const/SW = 128
+	var/const/NW = 64
+	var/const/SE = 32
+	var/const/NE = 16
```

Comment:
these exist already use those
```suggestion
```

Reply:
Yeah these do NOT already exist, these are four 8 bit flagging, having multiple directions at once. 4 bit flagging can't store `NORTH` and `WEST` at the same time, because `NORTH + WEST == NORTHWEST`, which is diagonal. These four variables store the diagonal directions at the same time as the cardinal directions in the same variable, `cable_surr`.
I accidentally clicked commit suggestion but i've had to undo that now because it broke.
You'll notice that later parts of the code use both kinds and swap between them.

Reply:
They've been renamed to `NORTHWEST_UNIQUE` and such for clarity